### ISSUE TITLE
=== ligature isn't evenly spaced at 13px

### ISSUE THREAD

**Issue by yishn** [JetBrains/JetBrainsMono#25]
At the recommended font size of `13px`, the `===` ligature doesn't look evenly spaced:

![=== ligature](https://user-images.githubusercontent.com/9217349/72509141-361e2480-3847-11ea-9dac-10ecfd4c998f.png)

**Comment by thepheer** [comment]
Yeah, this is the only issue I have. 15px seems to be the minimal font size where `===` looks nice in VSCode. But the font itself doesn't feel right overall on that size, so it's actually 16px where I'm happy. I'd love to use it on 14px or so.

Other than that, pretty soild font. 👍

**Comment by philippnurullin** [comment]
Yes, thats a problem. It appears on low dpi Win & Linux. Cant say the exact date when it will be fixed. Looks like it needs manual hinting. We will look in to that.
Thanks for feedback!

**Comment by aaronbell** [comment]
Definitely would recommend manual hinting if you’re looking to achieve good results at lower dpi screens :).

**Comment by gwk** [comment]
I encourage the developers to try to make this glyph look legible at low dpi 12px as well (and less if possible). This is the only issue I have seen so far that discourages me from using the font daily. Thanks for your efforts!

**Comment by EnderPicture** [comment]
This is my only gripe about this font so far, and since 13 is the suggested font size, I expected that it should look the best at that font size.
Other than that, very nice.

**Comment by vbsessa** [comment]
Same using 14px here.

**Comment by qgates** [comment]
=== stil looks imbalanced at a number of font sizes from 22 down. This issue should be reopened.

**Comment by SQReder** [comment]
Bump for this. Look broken at recommended 13px and 14px
![image](https://user-images.githubusercontent.com/1519753/81824987-bb3bf080-953e-11ea-8108-bf6faa7588b2.png)
![image](https://user-images.githubusercontent.com/1519753/81824741-7d3ecc80-953e-11ea-9c5d-06292e34d518.png)

**Comment by silkfire** [comment]
@philippnurullin What's the status of this issue? Did it enter the last release?

**Comment by philippnurullin** [comment]
Hi @silkfire , don't have a good solution for that yet.

**Comment by RosaWagner** [comment]
Try to have the exact same space between the bars? right now is it 140 and 136, at small ppm size, it can create a difference. Enlarge it even a little bit, like 180 or 200 units between each bar… it may help the rasteriser to render it better ?

**Comment by philippnurullin** [comment]
Ok, will explore in this direction.

**Comment by philippnurullin** [comment]
In the v2.200 we able to reproduce this issue only on Linux. So please test do you have any changes after updating.

**Comment by trygveaa** [comment]
Nice, now it's much improved at most font sizes. Here are my results for different font sizes between 6 and 20:

Looks good in 2.200, and didn't look good in 2.002:
10
11
14
15
17
18
19

Looks good in 2.200 and looked the same in 2.002:
8
12
20

Doesn't look good in 2.200, and looked the same in 2.002:
9

Looks different in 2.200 from 2.002, but none of the versions look good:
6
7
13

Looks worse in 2.200, looked good in 2.002:
16

This is on Arch Linux in the kitty terminal emulator on a 1920x1080 96 DPI display.

**Comment by philippnurullin** [comment]
Hi @trygveaa . Thanks for the detailed report. The 6-9 are not sizes to work comfortably with. So the real problem are 13, 16. I'll see what can be done further in the next update.

**Comment by trygveaa** [comment]
Yes, I agree. Personally I use 11, so I'm happy.

**Comment by philippnurullin** [comment]
Hi @trygveaa I want to tune the `===` more. Can you upload the screenshots with problematic rendering at 13, 16? It will be a great help.

**Comment by trygveaa** [comment]
Sure, here is 13:
![image](https://user-images.githubusercontent.com/601966/102609808-64c60e80-412c-11eb-9044-17657ef16109.png)

And 16:
![image](https://user-images.githubusercontent.com/601966/102609855-74ddee00-412c-11eb-8ca3-939858d51eb1.png)

This is with the ttf files currently in the repo, i.e. version 2.211.

**Comment by trygveaa** [comment]
For comparison, here is 16 from version 2.002, i.e. when it looked good:
![image](https://user-images.githubusercontent.com/601966/102610293-3bf24900-412d-11eb-8ff0-070b226305ca.png)

**Comment by philippnurullin** [comment]
Hi, please test the new version in the latest push. Thanks in advance!

**Comment by trygveaa** [comment]
I generated images for how it looks in sizes 8 to 20 in version 2.230, and ran some git bisects to compare old and newer versions.

In my opinion these are the ones that don't look as good as they have previously. This is the latest version a size looked good in, and the first version it looks worse in:
Sizes 8, 14 and 15 looked good in v2.225 (acab62a), but looks worse in v2.230 (f668e09).
Size 12 looked good in v2.211 (cb0730c), but looks worse in v2.220 (7445d1f) (same in v2.230 as in v2.220).
Size 16 looked good in v2.002 (0d9ba3f), looked a bit worse in 400d12f and changed again to a bit worse than that in v2.220 (7445d1f) (same in v2.230 as in v2.220).
Sizes 18 and 19 looked good in v2.211 (cb0730c), but looks slightly worse in v2.220 (7445d1f) and changed again to a bit worse than that in v2.230 (f668e09).

Size 13 looks good in 2.230, which is an improvement over previous versions.

Sizes 10, 11, 17 and 20 looked good previously, and still looks good in v2.230.

Size 9 didn't look good previously, and still doesn't.

Here's how it looks for me in v2.230:

Size 08:
![triple_equals_size_08_original](https://user-images.githubusercontent.com/601966/115585195-b396d700-a2cb-11eb-9a3b-9b3baff553a9.png)
Size 09:
![triple_equals_size_09_original](https://user-images.githubusercontent.com/601966/115585197-b42f6d80-a2cb-11eb-949a-76fe670ec68f.png)
Size 10:
![triple_equals_size_10_original](https://user-images.githubusercontent.com/601966/115585200-b42f6d80-a2cb-11eb-9c76-5c30dfd2d952.png)
Size 11:
![triple_equals_size_11_original](https://user-images.githubusercontent.com/601966/115585203-b4c80400-a2cb-11eb-8e51-f634968fe344.png)
Size 12:
![triple_equals_size_12_original](https://user-images.githubusercontent.com/601966/115585209-b5609a80-a2cb-11eb-8fee-e87aab6ed704.png)
Size 13:
![triple_equals_size_13_original](https://user-images.githubusercontent.com/601966/115585216-b5609a80-a2cb-11eb-8680-156accbdfd93.png)
Size 14:
![triple_equals_size_14_original](https://user-images.githubusercontent.com/601966/115585220-b5f93100-a2cb-11eb-999a-b6a7873dd0b3.png)
Size 15:
![triple_equals_size_15_original](https://user-images.githubusercontent.com/601966/115585222-b691c780-a2cb-11eb-87b5-117e6eee23fb.png)
Size 16:
![triple_equals_size_16_original](https://user-images.githubusercontent.com/601966/115585225-b691c780-a2cb-11eb-96ab-0fef913462b9.png)
Size 17:
![triple_equals_size_17_original](https://user-images.githubusercontent.com/601966/115585228-b72a5e00-a2cb-11eb-9bfe-a28475476fef.png)
Size 18:
![triple_equals_size_18_original](https://user-images.githubusercontent.com/601966/115585231-b7c2f480-a2cb-11eb-88fa-a43cf764adcb.png)
Size 19:
![triple_equals_size_19_original](https://user-images.githubusercontent.com/601966/115585236-b7c2f480-a2cb-11eb-96b5-50afed7004c3.png)
Size 20:
![triple_equals_size_20_original](https://user-images.githubusercontent.com/601966/115585240-b85b8b00-a2cb-11eb-8a81-5da22fc87aaf.png)

Scaled up:

Size 08:
![triple_equals_size_08_scaled](https://user-images.githubusercontent.com/601966/115585196-b42f6d80-a2cb-11eb-9a01-b73405544aae.png)
Size 09:
![triple_equals_size_09_scaled](https://user-images.githubusercontent.com/601966/115585198-b42f6d80-a2cb-11eb-97a1-e26f41dc49e4.png)
Size 10:
![triple_equals_size_10_scaled](https://user-images.githubusercontent.com/601966/115585201-b4c80400-a2cb-11eb-8cef-8de94be9f87b.png)
Size 11:
![triple_equals_size_11_scaled](https://user-images.githubusercontent.com/601966/115585205-b4c80400-a2cb-11eb-9469-f0136ccc184f.png)
Size 12:
![triple_equals_size_12_scaled](https://user-images.githubusercontent.com/601966/115585212-b5609a80-a2cb-11eb-94b7-b3a5ee57c545.png)
Size 13:
![triple_equals_size_13_scaled](https://user-images.githubusercontent.com/601966/115585218-b5f93100-a2cb-11eb-95ec-52a696f92ac8.png)
Size 14:
![triple_equals_size_14_scaled](https://user-images.githubusercontent.com/601966/115585221-b5f93100-a2cb-11eb-9019-ce15be7fdd5f.png)
Size 15:
![triple_equals_size_15_scaled](https://user-images.githubusercontent.com/601966/115585223-b691c780-a2cb-11eb-844c-29fb7f180c67.png)
Size 16:
![triple_equals_size_16_scaled](https://user-images.githubusercontent.com/601966/115585227-b72a5e00-a2cb-11eb-9697-91e0a91c1f8e.png)
Size 17:
![triple_equals_size_17_scaled](https://user-images.githubusercontent.com/601966/115585229-b72a5e00-a2cb-11eb-88db-563a2c9cd1f2.png)
Size 18:
![triple_equals_size_18_scaled](https://user-images.githubusercontent.com/601966/115585234-b7c2f480-a2cb-11eb-8158-96729ab52ead.png)
Size 19:
![triple_equals_size_19_scaled](https://user-images.githubusercontent.com/601966/115585239-b85b8b00-a2cb-11eb-9693-c92a51e7491d.png)
Size 20:
![triple_equals_size_20_scaled](https://user-images.githubusercontent.com/601966/115585242-b85b8b00-a2cb-11eb-8d7c-2af68c4f90b9.png)

**Comment by carlosala** [comment]
this continues to happen, any workaround?

**Comment by Calinou** [comment]
This should be fixed by 2.304: https://github.com/JetBrains/JetBrainsMono/releases/tag/v2.304
